File: nicolaspuebla_proyecto_final/src/main/java/com/example/nicolaspuebla_proyecto_final/repository/LocalityRepository.java
package com.example.nicolaspuebla_proyecto_final.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import com.example.nicolaspuebla_proyecto_final.model.dataModels.Locality;

/**
 * Repositorio que interactúa con la base de datos manejando las localidades.
 */
@Repository
public interface LocalityRepository extends JpaRepository<Locality, Long> {

}

File: nicolaspuebla_proyecto_final_android/app/src/main/java/com/example/nicolaspuebla_proyecto_final_android/data/repositories/LocalityRepository.kt
package com.example.nicolaspuebla_proyecto_final_android.data.repositories

import com.example.nicolaspuebla_proyecto_final_android.data.model.dto.LocalityListResponse
import com.example.nicolaspuebla_proyecto_final_android.data.remote.RetrofitInstance
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext

/**
 * Repositorio a través del cual los view models pueden ejecutar las peticiones necesarias al microservicio sobre la entidad Locality.
 */
class LocalityRepository {

    /**
     * Función que ejecuta la petición para obtener todas las localidades.
     *
     * @return Una lista con todas las localidades de la base de datos.
     */
    suspend fun getTeamLocalities(): LocalityListResponse?{
        return withContext(Dispatchers.IO){
            try {
                val call = RetrofitInstance.yellowLockerLocalityService.getLocalities()
                val response = call.execute()

                if(response.isSuccessful){
                    response.body()
                } else if(response.code() == 401){
                    throw Exception("401")
                } else {
                    throw Exception("Internal server error")
                }
            } catch (e:Exception){
                throw Exception(e.message)
            }
        }
    }
}